spec.html

```diff
@@ -8871,18 +8871,22 @@ <h1>[[Construct]] ( _argumentsList_, _newTarget_ )</h1>
     </emu-clause>
 
     <emu-clause id="sec-createbuiltinfunction" aoid="CreateBuiltinFunction">
-      <h1>CreateBuiltinFunction ( _steps_, _internalSlotsList_ [ , _realm_ [ , _prototype_ ] ] )</h1>
-      <p>The abstract operation CreateBuiltinFunction takes arguments _steps_ and _internalSlotsList_ (a List of names of internal slots) and optional arguments _realm_ and _prototype_. _internalSlotsList_ contains the names of additional internal slots that must be defined as part of the object. This operation creates a built-in function object. It performs the following steps when called:</p>
+      <h1>CreateBuiltinFunction ( _stepsOrClosure_, _internalSlotsList_ [ , _realm_ [ , _prototype_ ] ] )</h1>
+      <p>The abstract operation CreateBuiltinFunction takes arguments _stepsOrClosure_ and _internalSlotsList_ (a List of names of internal slots) and optional arguments _realm_ and _prototype_. _internalSlotsList_ contains the names of additional internal slots that must be defined as part of the object. This operation creates a built-in function object. It performs the following steps when called:</p>
       <emu-alg>
-        1. Assert: _steps_ is either a set of algorithm steps or other definition of a function's behaviour provided in this specification.
+        1. Assert: _stepsOrClosure_ is either an Abstract Closure, a set of algorithm steps or other definition of a function's behaviour provided in this specification.
         1. If _realm_ is not present, set _realm_ to the current Realm Record.
         1. Assert: _realm_ is a Realm Record.
         1. If _prototype_ is not present, set _prototype_ to _realm_.[[Intrinsics]].[[%Function.prototype%]].
-        1. Let _func_ be a new built-in function object that when called performs the action described by _steps_. The new function object has internal slots whose names are the elements of _internalSlotsList_.
+        1. Let _func_ be a new built-in function object that when called performs the action described by _stepsOrClosure_. The new function object has internal slots whose names are the elements of _internalSlotsList_.
         1. Set _func_.[[Realm]] to _realm_.
         1. Set _func_.[[Prototype]] to _prototype_.
         1. Set _func_.[[Extensible]] to *true*.
         1. Set _func_.[[ScriptOrModule]] to *null*.
+        1. If _stepsOrClosure_ is an Abstract Closure, then
+          1. Let _len_ be the largest number of named parameters in the definition of the Abstract Closure _stepsOrClosure_. Optional parameters (which are indicated with brackets: `[` `]`) or rest parameters (which are shown using the form &laquo;...name&raquo;) are not included in the parameter count.
```

Comment:
Abstract closures do not have optional parameters, so this could just be
```suggestion
          1. Let _len_ be number of parameters taken by _stepsOrClosure_.
```